File: API/Entities/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Project
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Status { get; set; }
        public int Progress { get; set; }
        public bool IsPublic { get; set; }
        public string Intro { get; set; }
        public string ProjectWith { get; set; }
        public string Description { get; set; }
        public string MainFeature { get; set; }
        public string Url { get; set; }
        public string GithubUrl { get; set; }
        public string FrontEnd { get; set; }
        public string BackEnd { get; set; }
        public string Database { get; set; }
        public string Deployement { get; set; }
        public DateTime? ProjectStarted { get; set; }
        public DateTime? ProjectEnded { get; set; }
        public AppUser AppUser { get; set; }
        public int AppUserId { get; set; }
        public ICollection<Photo> Images { get; set; }
        public ICollection<ProjectUser> TeamMembers { get; set; }
        public ICollection<ProjectLike> Likes { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedDate { get; set; }
        public ICollection<Skill> Skills { get; set; }
    }
}

File: client/src/app/_models/project.ts
import { Member } from './member';
import { Message } from './message';
import { Photo } from './photo';

export interface TeamMemberDto {
  id: number;
  username: string;
  name: string;
  photoUrl: string;
  city: string;
  country: string;
}

export interface Project {
  id?: number;
  name: string;
  isPublic: boolean;
  isCurrent: boolean;
  projectWith: string;
  description: string;
  mainFeature: string;
  url: string;
  githubUrl: string;
  frontEnd: string;
  backEnd: string;
  database: string;
  deployment: string;
  projectStarted: string;
  projectEnded: string;
  images: Photo[];
  user: Partial<Member>;
  teamMembers: TeamMemberDto[];
  status: number;
  progress: number;
}
